src/Core/Managed/Shared/Extensibility/Implementation/TelemetryProcessorChain.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using Microsoft.ApplicationInsights.Channel;
+
+    /// <summary>
+    /// Represents the TelemetryProcessor chain. 
+    /// </summary>
+    public sealed class TelemetryProcessorChain
+    {
+        private SnapshottingList<ITelemetryProcessor> telemetryProcessors = new SnapshottingList<ITelemetryProcessor>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TelemetryProcessorChain" /> class.
+        /// Marked internal, as clients should use TelemetryProcessorBuilder to build the processing chain.
```

Comment:
We should add this comment to the class summary as well - Use TelemetryProcessorChainBuilder to build the processing chain.